RecoPixelVertexing/PixelLowPtUtilities/bin/ClusterShapeAnalyzer.cc

```diff
@@ -0,0 +1,400 @@
+#include <cmath>
+
+#include <utility>
+#include <vector>
+#include <map>
+#include <fstream>
+#include <iostream>
+
+#include "TROOT.h"
+#include "TFile.h"
+#include "TH2F.h"
+#include "TH1F.h"
+#include "TNtuple.h"
+
+// Total loss, loss in one direction is Loss/2 
+#define Loss 0.005
+
+// FIXME
+#define MinEntries 100
+
+#define exMax 10
+#define eyMax 15
+
+using namespace std;
+
+/****************************************************************************/
+void printToFile(TH2F *h2, const char* fileName)
+{
+  char name[256];
+
+  sprintf(name,"%s.dat",fileName);
+  ofstream file(name);
+
+  for(int i = 1; i <= h2->GetNbinsX(); i++)
+  {
+    for(int j = 1; j <= h2->GetNbinsY(); j++)
+      file << " " << h2->GetXaxis()->GetBinLowEdge(i)
+           << " " << h2->GetYaxis()->GetBinLowEdge(j)
+           << " " << h2->GetBinContent(i,j)
+           << endl;
+
+      file << " " << h2->GetXaxis()->GetBinLowEdge(i)
+           << " " << h2->GetYaxis()->GetXmax()
+           << " 0" << endl;
+    file << endl;
+  }
+
+  for(int j = 1; j <= h2->GetNbinsY(); j++)
+    file << " " << h2->GetXaxis()->GetXmax()
+         << " " << h2->GetYaxis()->GetBinLowEdge(j)
+         << " 0" << endl;
+
+    file << " " << h2->GetXaxis()->GetXmax()
+         << " " << h2->GetYaxis()->GetXmax()
+         << " 0" << endl;
+
+  file.close();
+}
+
+/****************************************************************************/
+class Point 
+{
+  public:
+    float x[2]; // position
+    int w;      // point weight
+    int n;      // cluster index
+};
+
+class Center
+{
+  public:
+    float x[2]; // position
+    int n;      // cluster index
+    float area, lim[2][2];
+};
+
+/****************************************************************************/
+class ClusterAnalyzer
+{
+  public:
+	void analyze(const std::string& inputFileName, const std::string& outputFileName);
+
+  private:
+    float distance(Center& c, Point& p);
+    void getVectors(vector<Point>& points, TH1D * x[2][2]);
+    double getFraction(TH1D * line, double fraction);
+    void kMeans    (vector<Point>& points, vector<Center>& c);
+    void findLimits(vector<Point>& points, vector<Center>& c);
+    void printOut  (TH2F * histo, vector<Point>& points, vector<Center>& c, char* flag);
+    void process   (TH2F * histo,          vector<Center>& c, char* flag);
+    
+    static TH1D * line;
```

Comment:
there are no static methods in ```ClusterAnalyzer```. So, there doesn't seem to be a need to make this static and disturb the static analyzer.